tools/pyboard.py

```diff
@@ -509,10 +516,11 @@ def fs_get(self, src, dest, chunk_size=256):
                 f.write(data)
         self.exec_("f.close()")
 
-    def fs_put(self, src, dest, chunk_size=256):
+    def fs_put(self, src, dest, chunk_size=256, verbose_size=2048):
         self.exec_("f=open('%s','wb')\nw=f.write" % dest)
+        src_size = os.path.getsize(src)
         with open(src, "rb") as f:
-            while True:
+            for __ in _progress_loop(chunk_size, src_size, src_size > verbose_size):
```

Comment:
I'm not sure this will work correctly in all cases.  It's possible that `f.read(chunk_size)` returns less than the requested size, in which case this loop needs to run for longer.

Reply:
progress calc now uses len of read data instead of chunk size